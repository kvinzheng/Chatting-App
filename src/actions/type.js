export const LOGIN = "LOGIN";
export const POST_ROOM_MESSAGES = "POST_ROOM_MESSAGES";

export const SEARCH_PROFILE_PENDING = "SEARCH_PROFILE_PENDING";
export const SEARCH_PROFILE_FULFILLED = "SEARCH_PROFILE_FULFILLED";
export const SEARCH_PROFILE_REJECTED = "SEARCH_PROFILE_REJECTED";

export const FETCH_ROOMS_PENDING = "FETCH_ROOMS_PENDING";
export const FETCH_ROOMS_FULFILLED = "FETCH_ROOMS_FULFILLED";
export const FETCH_ROOMS_REJECTED = "FETCH_ROOMS_REJECTED";

export const FETCH_ROOM_DETAIL_PENDING = "FETCH_ROOM_DETAIL_PENDING";
export const FETCH_ROOM_DETAIL_FULFILLED = "FETCH_ROOM_DETAIL_FULFILLED";
export const FETCH_ROOM_DETAIL_REJECTED = "FETCH_ROOM_DETAIL_REJECTED";

export const FETCH_ROOM_MESSAGES_PENDING = "FETCH_ROOM_MESSAGES_PENDING";
export const FETCH_ROOM_MESSAGES_FULFILLED = "FETCH_ROOM_MESSAGES_FULFILLED";
export const FETCH_ROOM_MESSAGES_REJECTED = "FETCH_ROOM_MESSAGES_REJECTED";

export const UPDATE_ACTIVE_USER = "UPDATE_ACTIVE_USER";
