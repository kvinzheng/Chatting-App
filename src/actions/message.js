import {
  FETCH_ROOM_MESSAGES_PENDING,
  FETCH_ROOM_MESSAGES_FULFILLED,
  FETCH_ROOM_MESSAGES_REJECTED,
  POST_ROOM_MESSAGES,
} from "./type";
import baseUrl from "../api/baseUrl";

export const fetchSingleRoomMessages = (id) => {
  return async (dispatch) => {
    dispatch({
      type: FETCH_ROOM_MESSAGES_PENDING,
    });
    try {
      const { data } = await baseUrl.get(`/rooms/${id}/messages`);
      dispatch({
        type: FETCH_ROOM_MESSAGES_FULFILLED,
        payload: {
          id,
          data: data,
        },
      });
    } catch (e) {
      dispatch({
        type: FETCH_ROOM_MESSAGES_REJECTED,
      });
    }
  };
};

export const postMessageSingleRoom = (id, message) => {
  return {
    type: POST_ROOM_MESSAGES,
    payload: {
      id,
      data: message,
    },
  };
};
