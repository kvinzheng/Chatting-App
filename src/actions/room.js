import {
  FETCH_ROOMS_PENDING,
  FETCH_ROOMS_FULFILLED,
  FETCH_ROOMS_REJECTED,
  FETCH_ROOM_DETAIL_PENDING,
  FETCH_ROOM_DETAIL_FULFILLED,
  FETCH_ROOM_DETAIL_REJECTED,
  UPDATE_ACTIVE_USER
} from "./type";
import baseUrl from "../api/baseUrl";

export const fetchRooms = () => {
  return async (dispatch) => {
    dispatch({
      type: FETCH_ROOMS_PENDING,
    });
    try {
      const { data } = await baseUrl.get("/rooms");
      dispatch({
        type: FETCH_ROOMS_FULFILLED,
        payload: data,
      });
    } catch (e) {
      dispatch({
        type: FETCH_ROOMS_REJECTED,
      });
    }
  };
};

export const fetchSingleRoom = (id) => {
  return async (dispatch) => {
    dispatch({
      type: FETCH_ROOM_DETAIL_PENDING,
    });
    try {
      const { data } = await baseUrl.get(`/rooms/${id}`);
      dispatch({
        type: FETCH_ROOM_DETAIL_FULFILLED,
        payload: data,
      });
    } catch (e) {
      dispatch({
        type: FETCH_ROOM_DETAIL_REJECTED,
      });
    }
  };
};


export const updateActiveUsers = (users) => {
  return {
      type: UPDATE_ACTIVE_USER,
      payload: users
  };
};
