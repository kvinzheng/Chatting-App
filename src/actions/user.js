import { LOGIN } from "./type";

export const userLogin = (username) => {
  return {
    type: LOGIN,
    payload: { username, startTime: new Date() },
  };
};
