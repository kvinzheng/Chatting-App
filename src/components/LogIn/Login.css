.Login {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
  width: 100%;
}
.Login-content {
  width: 500px;
}
.Login-input {
  margin-bottom: 20px;
  box-sizing: border-box;
}

.Login-button-submit {
  color: white;
  background-color: #1877f2;;
  border: none;
  cursor: pointer;
  font-size: 16px;
  padding: 11px;
  border-radius: 5px;
  display: block;
  width: 100%;
}
