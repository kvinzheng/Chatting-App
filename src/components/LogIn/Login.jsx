import React, { useState } from "react";
import { connect } from "react-redux";
import { useHistory } from "react-router-dom";

import InputBox from "../../common/InputBox";
import { userLogin } from "../../actions/user";
import "./Login.css";

export const Login = ({ userLogin }) => {
  const [username, setUsername] = useState("");
  const history = useHistory();

  const handleSubmit = (event) => {
    event.preventDefault();
    if (!username) return;
    userLogin(username);
    history.push("/chat-rooms");
  };

  const handleEnterKeyDown = (event) => {
    if (event.key === "Enter") {
      handleSubmit(event);
    }
  };
  return (
    <div className="Login">
      <div className="Login-content">
        <InputBox
          name="username"
          className="Login-input"
          onChange={(e) => setUsername(e.target.value)}
          value={username}
          placeholder="Type your username..."
          onKeyDown={handleEnterKeyDown}
        />
        <button className="Login-button-submit" onClick={handleSubmit}>
          Join the Chat!
        </button>
      </div>
    </div>
  );
};

export const mapStateToProps = () => {
  return {};
};

export default connect(mapStateToProps, { userLogin })(Login);
