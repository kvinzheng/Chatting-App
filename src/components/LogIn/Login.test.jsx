import React from "react";
import { configure, mount } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { LOGIN } from "../../actions/type";
import { Login } from "./Login.jsx";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";

configure({ adapter: new Adapter() });

const userLogin = jest.fn();
jest.mock("react-router-dom", () => ({
  useHistory: () => ({
    push: jest.fn(),
  }),
}));

describe("Login Component", () => {
  const component = mount(<Login userLogin={userLogin} />);
  it("should render with testing data ", () => {
    expect(component).toMatchSnapshot();
  });
  it("shouldn't go to another page when click given no input", () => {
    component.find(".Login-button-submit").simulate("click");
    expect(global.window.location.pathname).toEqual("/");
  });
});
