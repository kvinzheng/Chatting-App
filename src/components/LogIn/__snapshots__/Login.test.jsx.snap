// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component should render with testing data  1`] = `ReactWrapper {}`;
